solutions/silver/cf-104536F.mdx

```diff
@@ -0,0 +1,82 @@
+---
+id: cf-104536F
+source: CF
+title: Minimize the Diameter
+author: Kenneth Karaisz Jr.
+---
+
+## Explanation
+
+Let's consider the longest possible path the added edge could create. One way to find this path would be to use the two farthest nodes on each tree from the nodes the added edge connects.
+Therefore, on each tree, the optimal node to connect the edge to is the node that has the shortest possible path to the farthest node.
+
+It turns out that this node is always the median of the tree's diameter. To help understand this, we can visualize the tree with its diameter as a horizontal line, and the other nodes
```

Comment:
wdym by "median" of the tree's diameter

Reply:
The median is the middle node. It’s a pretty common word, I think it’s fine.

Reply:
@ryanchou-dev opinion? i've never seen median used this way before

Reply:
just say middle if it means the same thing... median can get confusing imo